subprojects/core-api/src/main/java/org/gradle/api/attributes/AttributeContainer.java

```diff
@@ -57,6 +57,17 @@
      */
     <T> AttributeContainer attribute(Attribute<T> key, T value);
 
+    /**
+     * Sets an attribute value by providing the string representation of the value.
+     * It is not allowed to use <code>null</code> as an attribute value.
```

Comment:
:x: Javadocs should mention that coercion is done automatically.

Reply:
I extended the javadoc and added a test case.